### PR TITLE
Remove Milestones

### LINKED ISSUE #737: Remove all outdated code referring to the getMilestones()
Remove the methods around getMilestones() as the project went in a different direction

### PR DESCRIPTION
**Pull request by mmschaefer**
Fix #737 - remove the unused Milestone tracking for overall test progress.  We went in a different direction.

### CODE REVIEW

File: src-test/org/etools/j1939_84/ui/UserInterfacePresenterTest.java [hunk 1]
```diff
@@ -641,7 +642,7 @@ public void testOnStopButtonClickedWithNoController() {
 
     @Test
     public void testOnStopButtonClickedWithStoppedController() {
-        when(overallController.isActive()).thenReturn(false);
+        when(Controller.isActive()).thenReturn(false);
```
Comment (gumbelmj) [inline src-test/org/etools/j1939_84/ui/UserInterfacePresenterTest.java:645]: You can't mock statics.  I'm surprise the tests don't fail here.

File: src/org/etools/j1939_84/modules/ReportFileModule.java [hunk 2]
```diff
@@ -81,28 +79,6 @@ public void addOutcome(int partNumber, int stepNumber, Outcome outcome, String m
...
-    @Override
-    public void endPart(PartResult partResult) {
-        // onResult(NL);
-        // onResult("End Part " + partResult);
-    }
-
-    @Override
-    public void endStep(StepResult stepResult) {
-        // onResult(NL);
-        summaryModule.endStep(stepResult);
```
Comment (gumbelmj) [inline src/org/etools/j1939_84/modules/ReportFileModule.java:103]: I think this needs to remain

File: src/org/etools/j1939_84/modules/SummaryModule.java [hunk 3]
```diff
@@ -38,16 +38,6 @@ public void addOutcome(int partNumber, int stepNumber, Outcome outcome, String m
         stepResult.addResult(new ActionOutcome(outcome, message));
     }
 
-    public void beginPart(PartResult partResult) {
-        getPartResults().add(partResult);
-    }
-
-    public void endStep(StepResult stepResult) {
-        if (stepResult.getOutcome() == Outcome.INCOMPLETE) {
```
Comment (gumbelmj) [inline src/org/etools/j1939_84/modules/SummaryModule.java:46]: I think the `SummaryModule` needs to remain or we need to move this logic to the PartController to add a `PASS` Outcome when the `StepController` has completed without any other outcome.

File: src-test/org/etools/j1939_84/controllers/part01/Part01Step16ControllerTest.java [hunk 4]
```diff
@@ -232,7 +232,8 @@ public void testNonOBDMilOn() {
                                         "6.1.16.2.c - Non-OBD ECU Engine #1 (0) did not report MIL off or not supported");
 
         assertEquals("", listener.getMessages());
-        assertEquals("", listener.getResults());
+        String expected = "";
```
Comment (gumbelmj) [inline src-test/org/etools/j1939_84/controllers/part01/Part01Step16ControllerTest.java]: This change should be reverted

File: src-test/org/etools/j1939_84/controllers/part03/Part03Step04ControllerTest.java [hunk 5]
```diff
@@ -489,6 +489,7 @@ public void testNoFailures() {
         runTest();
 
         assertEquals("", listener.getMessages());
+
```
Comment (gumbelmj) [inline src-test/org/etools/j1939_84/controllers/part03/Part03Step04ControllerTest.java]: No need to commit this file

File: src-test/org/etools/j1939_84/controllers/part03/Part03Step15ControllerTest.java [hunk 6]
```diff
@@ -174,7 +173,7 @@ public void testFailure() {
 
         assertEquals("", listener.getResults());
         assertEquals("", listener.getMessages());
-        assertEquals("", listener.getMilestones());
+
```
Comment (gumbelmj) [inline src-test/org/etools/j1939_84/controllers/part03/Part03Step15ControllerTest.java:176]: nit: extra lines here and below

**Review by gumbelmj — CHANGES_REQUESTED** [review]

File: src/org/etools/j1939_84/modules/SummaryModule.java (hunk 3 above)
Comment (mmschaefer) [reply src/org/etools/j1939_84/modules/SummaryModule.java:46]: I moved the logic into the PartController to make life simpler.

File: src/org/etools/j1939_84/modules/ReportFileModule.java (hunk 2 above)
Comment (mmschaefer) [reply src/org/etools/j1939_84/modules/ReportFileModule.java:103]: Moved it to the stepController as suggested below.

**Review by gumbelmj — APPROVED** [review]

**Review by gumbelmj — APPROVED** [review]

**Pull request merged**